arcgis-ios-sdk-samples/Content Display Logic/Controllers/OptionsTableViewController.swift

```diff
@@ -31,30 +31,66 @@ class OptionsTableViewController: UITableViewController {
     /// change handler.
     /// - Parameters:
     ///   - options: The options displayed by the view controller.
     ///   - selectedIndex: The index of the currently selected option or `nil`.
+    ///   - allowsEmptySelection: A bool to determine if empty selection is allowed. Defaulted to false.
     ///   - onChange: A closure called when the selected option has changed.
-    init(options: [Option], selectedIndex: Int?, onChange: @escaping (Int) -> Void) {
+    init(options: [Option], selectedIndex: Int?, allowsEmptySelection: Bool = false, onChange: @escaping (Int?) -> Void) {
         self.options = options
         self.selectedIndex = selectedIndex
+        self.allowsEmptySelection = allowsEmptySelection
         self.onChange = onChange
         super.init(nibName: nil, bundle: nil)
     }
     
-    /// Creates a new instance with the given labels, selected index, and
+    /// Creates a new instance with the options, selected index, empty selection bool, and selection
+    /// change handler.
+    /// - Parameters:
+    ///   - options: The options displayed by the view controller.
+    ///   - selectedIndex: The index of the currently selected option or `nil`.
+    ///   - onChange: A closure called when the selected option has changed.
+    convenience init(options: [Option], selectedIndex: Int, onChange: @escaping (Int) -> Void) {
+        self.init(options: options, selectedIndex: selectedIndex, allowsEmptySelection: false, onChange: { onChange($0!) })
+    }
+    
+    /// Creates a new instance with the given labels, selected index, empty selection bool, and
+    /// selection change handler.
+    /// - Parameters:
+    ///   - labels: An array of labels for the options.
+    ///   - selectedIndex: The index of the currently selected option or `nil`.
+    ///   - onChange: A closure called when the selected option has changed.
+    convenience init(labels: [String], selectedIndex: Int, onChange: @escaping (Int) -> Void) {
+        let options = labels.map { Option(label: $0) }
+        self.init(options: options, selectedIndex: selectedIndex, allowsEmptySelection: false, onChange: { onChange($0!) })
+    }
+    
+    /// Creates a new instance with the given labels, selected index, empty selection bool, and
     /// selection change handler.
     /// - Parameters:
     ///   - labels: An array of labels for the options.
     ///   - selectedIndex: The index of the currently selected option or `nil`.
     ///   - onChange: A closure called when the selected option has changed.
     convenience init(labels: [String], selectedIndex: Int?, onChange: @escaping (Int) -> Void) {
         let options = labels.map { Option(label: $0) }
-        self.init(options: options, selectedIndex: selectedIndex, onChange: onChange)
+        self.init(options: options, selectedIndex: selectedIndex, allowsEmptySelection: false, onChange: { onChange($0!) })
```

Comment:
```suggestion
        self.init(options: options, selectedIndex: selectedIndex, onChange: { onChange($0!) })
```

If we set a default value for `allowsEmptySelection` in the base constructor, then we can omit it in the convenience ones. However it might be good to keep it here so the `convenience init`s echo each other.

Reply:
Hm, yeah I'm afraid it might cause confusion if we omit `allowsEmptySelection` in some but leave it in others. I'll leave this open for discussion.